### ISSUE TITLE
CountVectorizer — convert text into bag-of-words feature vectors

### ISSUE THREAD

**Issue by DeathSurfing** [featrs/featrs#76]
## Description

`CountVectorizer` converts text documents (Polars String column) into a **bag-of-words** feature matrix. It tokenizes each document, learns a vocabulary across all training documents, and produces a numeric table where each row is a document and each column is a vocabulary term, with cell values being the count of that term in that document.

This is the equivalent of scikit-learn's [`CountVectorizer`](https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.CountVectorizer.html).

## Proposed API

```rust
pub struct CountVectorizer {
    fitted: bool,
    column: Option<String>,                // single column for now
    tokenizer: Tokenizer,                  // see variants below
    ngram_range: (usize, usize),            // default: (1, 1) — unigrams
    max_features: Option<usize>,            // cap on vocabulary size (most frequent)
    min_df: usize,                         // min document frequency, default: 1
    max_df: f64,                            // max document frequency as fraction (0..1), default: 1.0 (no cap)
    stop_words: Option<Vec<String>>,
    lowercase: bool,                        // default: true
    strip_accents: bool,                    // default: false (would require unicode deps; defer)
    vocabulary: Option<HashMap<String, u32>>, // learned word → column index
}

pub enum Tokenizer {
    Whitespace,                            // split on whitespace (fast, simple)
    WordRegex(String),                     // user-supplied regex pattern
}

impl CountVectorizer {
    pub fn new() -> Self;
    pub fn column(mut self, c: &str) -> Self;
    pub fn tokenizer(mut self, t: Tokenizer) -> Self;
    pub fn ngram_range(mut self, lo: usize, hi: usize) -> Self;
    pub fn max_features(mut self, n: usize) -> Self;
    pub fn min_df(mut self, n: usize) -> Self;
    pub fn max_df(mut self, p: f64) -> Self;
    pub fn stop_words(mut self, words: Vec<String>) -> Self;
    pub fn lowercase(mut self, b: bool) -> Self;
}

impl Default for CountVectorizer { ... }

impl Fit<DataFrame, Output=()> for CountVectorizer { ... }
impl Transform<DataFrame, Output=DataFrame> for CountTransformer { ... }
```

## Implementation Guidance

1. Create `src/preprocessing/text/count_vectorizer.rs` (new submodule `text/`)
2. In `fit`:
   - Validate non-empty input, `column` specified and is String
   - For each document (row in the column):
     - Optionally `.to_lowercase()` (per `lowercase`)
     - Tokenize:
       - **Whitespace**: `.split_whitespace().collect()`
       - **WordRegex**: precompile regex (reuse #21 StringCleaner dep); match all occurrences
     - Filter stop_words
     - Generate n-grams in `ngram_range.0..=ngram_range.1` by joining adjacent tokens with `" "` (or `_`)
   - Build `vocabulary: HashMap<String, u32>` — across all documents:
     - Count document frequency (DF) per term
     - Filter terms: `min_df <= df` and `df <= max_df * n_docs`
     - If `max_features` set: keep top-N by **term frequency across all docs** (DF ties → alphabetical)
     - Assign column index `0..V`
   - Store vocabulary
   - Set `fitted = true`
3. In `transform`:
   - For each document, tokenize (same way) and produce a **sparse** count vector `Vec<Option<u32>>` of length `V`. Tokenize → normalize → map each token to vocabulary index → accumulate counts (in dense `Vec<u32>` of size V, only counting hits; convert to `Vec<Option<f64>>` or to sparse)
   - Build a DataFrame with V Float64 columns named by vocabulary term
   - Output dtype: Float64 (for compatibility with downstream) or UInt32; Float64 to match standard pattern
4. Register module, add to prelude

## Edge Cases

- **Unknown terms at transform**: silently dropped (not in vocabulary)
- **Empty string document**: row of zeros
- **Max_df too high** (e.g., 1.0): no filtering; too-low (e.g., 0.01): rare-only — verify behavior
- **Min_df = 0**: error or treat as 1
- **Ngram range (1, 2)**: unigrams + bigrams
- **Stop words applied before n-gram formation**: standard — needs to be documented
- **Lowercase applied before tokenization**: also standard
- **All documents are empty**: vocabulary is empty → output is empty df
- **Single document**: df count == 1 for every term
- **Very large vocabulary (`V > 10000`)**: output DataFrame may be huge (sparse storage trade-off). Document that dense columns are used; future sparse support is in LazyFrame issue #33
- **Unicode**: rust's `.to_lowercase()` handles UTF-8 correctly. For `strip_accents`, defer this option to a later feature (requires a unicode-normalization crate)
- **冲突 with regex pattern**: error at fit if invalid regex

## Testing

- Small corpus: `["the cat sat", "the dog ran", "cat and dog"]` → vocabulary of {the, cat, sat, dog, ran, and}, DF {2, 2, 1, 2, 1, 1}
- Counts: `[[1, 1, 1, 0, 0, 0], [1, 0, 0, 1, 1, 0], [0, 1, 0, 1, 0, 1]]`
- max_df=0.5: drop terms in >50% docs ("the" likely removed)
- min_df=2: keep only terms in >= 2 docs
- max_features=2: keep top-2 most frequent
- ngram_range=(1, 2): both ugram + bgram columns
- Stop words: ["the"] removed from vocabulary
- Unknown term at transform time ignored
- Empty doc → all-zero row
- Not-fitted error

## Dependencies

- **New dependency**: `regex` (also needed by #21 StringCleaner; share)
- **Optional**: a unicode-normalization crate for `strip_accents` (defer to a future issue)

## Related Issues

- #31 — `TFIDFVectorizer` (builds on top of `CountVectorizer` — share vocabulary, share tokenizer)
- #32 — `CharacterNGramVectorizer` (sibling — different tokenizer)
- #21 — `StringCleaner` (shares `regex` dependency)
- #38 — FeatureNamesOut (critical — column names from vocabulary)
- #33 — LazyFrame Support (future sparse storage for large vocabularies)
- #26 — `ahash` for FeatureHasher (could use hashing trick to bound vocabulary)
